src/SFML/Graphics/Texture.cpp

```diff
@@ -100,6 +100,43 @@ Texture::~Texture()
 }
 
 
+////////////////////////////////////////////////////////////
+Texture::Texture(Texture&& source) noexcept :
+m_size(source.m_size),
+m_actualSize(source.m_actualSize),
+m_texture(source.m_texture),
+m_isSmooth(source.m_isSmooth),
+m_sRgb(source.m_sRgb),
+m_isRepeated(source.m_isRepeated),
+m_pixelsFlipped(source.m_pixelsFlipped),
+m_fboAttachment(source.m_fboAttachment),
+m_hasMipmap(source.m_hasMipmap),
+m_cacheId(TextureImpl::getUniqueId())
```

Comment:
Is this correct, or do we want to reuse the same cache id as the moved-from texture? Not familiar with how the cache really works.

Reply:
We had problems before with trying to be smart about reducing the number of texture bind calls we did. Since this didn't really work out, we always assign new ids to force a rebind any time the object changes, even if the underlying texture object was moved.

Reply:
That's the main question of why #2276 wasn't merged yet 😅

Reply:
> We had problems before with trying to be smart about reducing the number of texture bind calls we did. Since this didn't really work out, we always assign new ids to force a rebind any time the object changes, even if the underlying texture object was moved.

Should we maybe add a code comment to clarify this for future readers? 🙂